spec/features/course/assessment/submission/log_spec.rb

```diff
@@ -31,31 +33,33 @@
         within find('tr', text: protected_assessment.title) do
           click_link 'Attempt'
         end
+        wait_for_page
 
-        submission = protected_assessment.submissions.last
-        expect(submission.logs.count).to equal(1)
-        expect(submission.logs.last.valid_attempt?).to be(true)
+        expect(protected_submission.logs.count).to equal(1)
+        expect(protected_submission.logs.last.valid_attempt?).to be(true)
 
         expect do
-          visit edit_course_assessment_submission_path(course, protected_assessment, submission)
-        end.not_to(change { submission.logs.count })
+          visit edit_course_assessment_submission_path(course, protected_assessment, protected_submission)
+        end.not_to(change { protected_submission.logs.count })
 
         # Logout and login again and visit the same submission
         logout
         login_as(user)
         wait_for_page
 
         expect do
-          visit edit_course_assessment_submission_path(course, protected_assessment, submission)
-        end.to change { submission.logs.count }.by(1)
-        expect(submission.logs.last.valid_attempt?).to be(false)
+          visit edit_course_assessment_submission_path(course, protected_assessment, protected_submission)
+        end.to change { protected_submission.logs.count }.by(1)
+        expect(protected_submission.logs.last.valid_attempt?).to be(false)
 
         expect do
-          fill_in 'session_password', with: protected_assessment.session_password
-          click_button I18n.t('course.assessment.sessions.new.continue')
+          fill_in 'password', with: protected_assessment.session_password
+          click_button('Submit')
           wait_for_page
-        end.to change { submission.logs.count }.by(1)
-        expect(submission.logs.last.valid_attempt?).to be(true)
+        end.to change { protected_submission.logs.count }.by(1)
+
+        wait_for_job
```

Comment:
What job?

Reply:
it's actually to wait for those authentication job (when you clicked on `Submit` button) to finish completely before checking the logs. I moved the `wait_for_page` from inside the nest to after that line (for some reason, the count is updated but the last log registered was still the second last one)

Reply:
Is a job run in the backend to authenticate the password?

Reply:
no, it's run in the frontend
I just understand completely what this `wait_for_job` entails. So instead, I think the right one is to wait for the page to be fully rendered after the submit button is clicked, hence I changed it into `wait_for_page`.